File: app/schemas/budget.py
from pydantic import BaseModel
from decimal import Decimal
from datetime import datetime

class BudgetCreate(BaseModel):
    category_id: int | None = None
    month: int
    year: int
    limit_amount: Decimal

class BudgetUpdate(BaseModel):
    limit_amount: Decimal | None = None

class BudgetResponse(BaseModel):
    id: int
    category_id: int | None = None
    month: int
    year: int
    limit_amount: Decimal
    spent: Decimal
    remaining: Decimal
    percentage_used: float
    is_exceeded: bool
    created_at: datetime

    class Config:
        from_attributes = True
        

File: frontend/src/types/budget.ts
// Mirrors app/schemas/budget.py. spent/remaining/percentage_used/is_exceeded
// are calculated live by the backend on every read, not stored.

export interface BudgetResponse {
  id: number;
  category_id: number | null; // null = an "overall" budget covering all categories
  month: number;
  year: number;
  limit_amount: string;
  spent: string;
  remaining: string;
  percentage_used: number;
  is_exceeded: boolean;
  created_at: string;
}

export interface BudgetCreatePayload {
  category_id: number | null;
  month: number;
  year: number;
  limit_amount: string;
}

export interface BudgetUpdatePayload {
  limit_amount: string;
}
